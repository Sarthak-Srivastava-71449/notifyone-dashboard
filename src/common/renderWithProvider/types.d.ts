/*
    Auth User Types
*/

interface IUser {
  loading: boolean;
  error: string;
  success: boolean | number;
  name: string;
  tokens: {
    accessToken: string;
  };
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
  roles: IRoles;
}

type UserAction = {
  type?: string;
  payload?: IUser;
};

type UserActionDispatchType = (args: UserAction) => UserAction | any;

interface AxiosGoogleResponse {
  auth_token: string;
  auth_token_labs: string;
  token: string;
  username: string;
}

interface UserInfoResponse {
  roles: IRoles;
  user_info: {
    email_id: string;
    contact_number: string;
    email_verified: boolean;
  };
}
