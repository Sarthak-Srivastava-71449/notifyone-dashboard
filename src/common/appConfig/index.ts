const AppConfig: any = process.env.communication;

export default AppConfig;
