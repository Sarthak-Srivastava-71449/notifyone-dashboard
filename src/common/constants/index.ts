export const COMMUNICATION_TYPE = {
  All: 'All',
  SMS: 'SMS',
  Whatsapp: 'Whatsapp',
  Transactional_Push_notification: 'Transactional Push Notification',
  Email: 'Email',
};

export const MAX_INPUT_LIMIT = 30;

export const AUTH_ROLES = 'LARA_ADMIN';

export const EVENT_FETCH_LIMIT = 50;

// Navigation
export const COMS_TEMPLATE_NAVIGATE = '/communication/template';
