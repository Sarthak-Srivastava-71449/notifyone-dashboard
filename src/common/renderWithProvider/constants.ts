export const mockEmailEventDetails = {
  actions: 1,
  app_name: 'off',
  description: 'ccq_orders_time_reporting',
  event_id: 207,
  event_name: 'ccq_orders_time_reporting',
  event_text: 'p PFA',
  event_type: 'Email',
  id: 183,
  includes: [],
  name: 'ccq_orders_time_reporting',
  subject: 'CCQ Reporting',
  triggers_limit: -1,
  updated_by: 'rohit.taneja@1mg.com',
};

export const mockSmsEventDetails = {
  actions: 1,
  app_name: 'Doctors',
  event_name: 'user_cancelled_appointment',
  event_text:
    'Dear {{patient}},\nYour consultation with {{doctor_name}} at {{time}}, {{date}} has been cancelled.\nRefund has been initated to your source account. It will reflect within 5-7 working days. We are sorry to see you go.\nThanks. -Tata 1mg',
  event_type: 'SMS',
  id: 504,
  triggers_limit: -1,
};

export const mockPushNotificationEventDetails = {
  actions: 1,
  app_name: 'offer_service',
  event_name: 'last_days_before_expiry',
  event_text: '{{body}}',
  event_type: 'Transactional Push Notification',
  id: 83,
  image: null,
  target: 'DYNAMIC TARGET',
  title: '{{title}}',
  triggers_limit: -1,
  updated: 1629096532,
  updated_by: 'lara_service_user',
};

export const mockWhatsappEventDetails = {
  actions: 0,
  app_name: 'diagnostics',
  event_name: 'report_discripency_whatsapp',
  event_text: '32',
  event_type: 'Whatsapp',
  id: 180,
  triggers_limit: -1,
};
export const mockRavenRootNodeEvents = {
  result: [
    {
      node_details: {
        sub_questions: [
          {
            sub_questions: [
              {
                updated_by: 'v-animesh.singh@1mg.com',
                event: null,
                required_fields: ['free_text'],
                emails: null,
                screen_text: null,
                optional_fields: null,
                type: 'FORM',
                node_name: '1.161.159',
                id: '1.161.159',
                header: 'Feedback',
                node_action: 73,
                response:
                  '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
                name: 'What is 1mg?',
                ticket: {
                  status: 'OPEN',
                  priority: 'MEDIUM',
                  updated_by: 'initialdbsetup@1mg.com',
                  id: 3,
                },
              },
              {
                updated_by: 'ankita.gupta@1mg.com',
                event: null,
                required_fields: null,
                emails: null,
                screen_text:
                  'You can easily reach out to us via email: <br>\n\n~ Any query related to Medicine orders, please click on <b><a href=https://www.1mg.com/order/history =>Order History</a> </b> then on Need Help. <br>\n~ Any query related to Lab Tests, please click on <b><a href=https://www.1mg.com/labs?utm_source=1mg&utm_medium=jewel&utm_campaign=labsgrowth =>Lab Tests </a> </b> then on Need Help on the right side corner. <br>\n~ Any query related to Doctor Consultation, click here <b><a href=https://www.1mg.com/online-doctor-consultation =>Online Consultation</a> </b>',
                optional_fields: null,
                type: 'INFO',
                node_name: '1.161.168',
                id: '1.161.168',
                header: 'How to contact us?',
                node_action: 176,
                response:
                  '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id within next 24 hours.',
                name: 'How to contact 1mg?',
                ticket: {
                  status: 'OPEN',
                  priority: 'LOW',
                  updated_by: 'initialdbsetup@1mg.com',
                  id: 2,
                },
              },
            ],
          },
        ],
        node_name: '466',
        id: '466',
        node_action: null,
        name: 'Cancellation Initiated',
        updated_by: 'gaurav.singh@1mg.com',
      },
    },
  ],
  success: true,
};

export const mockFetchRavenActionsNode = {
  result: [
    {
      event: null,
      application: 'PHARMACY',
      required_fields: null,
      screen_text:
        'If you are not 100% satisfied with your purchase for any reason, you may initiate a return request for the delivered products. Make sure that the products are unused and in their original packaging.<br><br>Once your return request is received and verified your refund will be processed and automatically applied to your original method of payment within 7 days. <br><br>We encourage you to please check the package carefully before you accept delivery.<br><br>Know more about our <a href=https://tdgx9.app.goo.gl/?link=https://www.1mg.com/return-policy&al=onemg-web://www.1mg.com/return-policy&ibi=com.aranoah.healthkart&ifl=www.1mg.com/return-policy&apn=com.aranoah.healthkart.plus&afl=www.1mg.com/return-policy>Return Policy</a>',
      emails: null,
      optional_fields: null,
      type: 'INFO',
      id: 42,
      header: '1mg Return policy',
      response:
        '<b>Thank You!</b><br>We have received your message and will be in touch over your registered email id as soon as possible.',
      updated_by: 'initialdbsetup@1mg.com',
      ticket: 1,
    },
  ],
  success: true,
};

export const mockFetchRavenTicket = {
  result: [
    {
      status: 'OPEN',
      priority: 'HIGH',
      updated_by: 'initialdbsetup@1mg.com',
      id: 1,
    },
    {
      status: 'OPEN',
      priority: 'LOW',
      updated_by: 'initialdbsetup@1mg.com',
      id: 2,
    },
    {
      status: 'OPEN',
      priority: 'MEDIUM',
      updated_by: 'initialdbsetup@1mg.com',
      id: 3,
    },
    {
      status: 'OPEN',
      priority: 'URGENT',
      updated_by: 'tushar.bansal@1mg.com',
      id: 4,
    },
  ],
  success: true,
};

export const mockFetchRavenMetaData = {
  result: {
    type: [
      {
        type: 'FORM',
        id: '1',
      },
      {
        type: 'INFO',
        id: '2',
      },
      {
        type: 'INFO_WITH_WRITE_TO_US',
        id: '3',
      },
      {
        type: 'WEBVIEW',
        id: '4',
      },
    ],
    fields: [
      {
        id: '1',
        fields: 'free_text',
      },
      {
        id: '2',
        fields: 'skus',
      },
      {
        id: '3',
        fields: 'attachments',
      },
    ],
  },
  success: true,
};

export const mockRavenNodes = {
  result: [
    {
      node_action: 177,
      updated_by: 'v-hritik.raj@1mg.com',
      application: 'PHARMACY',
      id: 473,
      name: 'eneter',
    },
    {
      node_action: 383,
      updated_by: 'deepanshu.kapoor@1mg.com',
      application: 'PHARMACY',
      id: 472,
      name: 'Tata NeuCoins le lo ji',
    },
  ],
  success: true,
};

export const mockCreateNodeEvent = {
  result: null,
  success: true,
};

export const mockGetLinkNodeEvents = {
  result: [
    {
      node_action: 88,
      name: 'I was overcharged for a product',
      application: 'PHARMACY',
      id: 89,
      updated_by: 'initialdbsetup@1mg.com',
    },
    {
      node_action: 88,
      name: 'I was overcharged for a product',
      application: 'PHARMACY',
      id: 68,
      updated_by: 'initialdbsetup@1mg.com',
    },
    {
      node_action: 88,
      name: 'Amount overcharged',
      application: 'PHARMACY',
      id: 48,
      updated_by: 'ankita.gupta@1mg.com',
    },
  ],
  success: true,
};
export const mockSendNodeLink = {
  result: null,
  success: true,
};
export const mockInactiveNodeLinks = {
  result: [
    '59.60',
    '1.161.462',
    '1.161.4.12',
    '1.161.4.13',
    '1.161.4.14',
    '1.161.4.17',
    '1.161.8.26',
    '1.161.8.27',
    '1.161.8.25',
    '1.161.6.22',
    '1.161.6.24',
    '1.161.6.23',
    '1.5.19',
    '151.107',
    '151.99',
    '151.101',
    '150.83',
    '151.96',
    '150.85',
    '150.85.91',
    '150.85.92',
    '150.85.93',
    '150.86.94',
    '150.86.95',
    '150.86',
    '150.87',
    '59.65',
    '150.82',
    '1.9.21',
    '58.38.55',
    '58.38.56',
    '58.38.57',
    '58.37.53',
    '58.37.54',
    '58.37',
    '58.34.41',
    '58.34.42',
    '58.34.43',
    '58.34.44',
    '58.34.45',
    '58.34.46',
    '58.34.47',
    '58.36.50',
    '58.36.51',
    '59.64.73',
    '59.64.74',
    '59.64.75',
    '59.62',
    '59.64',
    '148.76',
    '148.77.81',
    '148.77',
    '148.88',
    '1.161.462.460',
    '152.112.119',
    '154.143',
    '1.161.4.160',
    '1.161.2',
    '503.498.497',
    '1.161.6',
    '1.9.15',
    '1.9.18',
    '1.9.7',
    '1.163.164',
    '1.163.165',
    '1.5.20',
    '1.172',
    '152.112',
    '152.121',
    '152.112.120',
    '152.112.122',
    '152.113',
    '1.9.29',
    '152.116.125',
    '152.116.126',
    '152.116.127',
    '152.119',
    '152.111',
    '152.116',
    '151.101.109',
    '154.144',
    '151.98',
    '150.86.167',
    '150.90',
    '59.61.66',
    '59.61.67',
    '59.62.68',
    '59.62.69',
    '59.63.70',
    '59.63.71',
    '148.78',
    '148.166',
    '154.142',
    '155.145',
    '155.146',
    '229.220',
    '235.230',
    '246.236',
    '151.97.3',
    '253.247',
    '259.254',
    '1.229.220',
    '58.383',
    '263.260',
    '270.264',
    '271.270',
    '271.270.264',
    '280.272',
    '284.281',
    '288.285',
    '294.289',
    '1.307.306',
    '299.295',
    '307.306',
    '1.161.306',
    '311.309',
    '316.312',
    '1.470',
    '1.161.473',
    '1.471',
    '152.112.118',
    '1.278',
    '466.379',
    '151.97.103',
    '151.97.104',
    '151.101.108',
    '151.101.110',
    '321.317',
    '323.322',
    '1.161.455',
    '338.335',
    '151.97',
    '328.326',
    '333.330',
    '342.339',
    '347.343',
    '359.357',
    '352.349',
    '363.360',
    '367.364',
    '1.161.455.454',
    '150.378',
    '150.381',
    '1.383',
    '1.382',
    '58.381',
    '58.381.378',
    '1.380',
    '1.369',
    '1.378',
    '1.161.381.379',
    '1.161.4.379',
    '1.161.4.359',
    '1.161.4.356',
    '1.161.4.347.343',
    '1.474',
    '477.473',
    '479.453',
    '483.465',
    '153.134.383',
    '491.471',
    '1.491',
    '1.304',
    '1.283',
    '1.306',
    '1.42',
    '493.492',
    '493.492.491',
    '496.477',
    '499.495',
    '500.493',
    '150.381.378',
    '152.380',
    '58.378',
    '58.381.379',
    '58.370',
    '1.375',
    '58.375',
    '148.355',
    '1.347',
    '1.161.4.359.358',
    '1.161.4.358',
    '1.161.4.347',
    '474.472',
    '476.473',
    '478.473',
    '484.471',
    '485.468',
    '450.452.380',
    '1.491.471',
    '372.375',
    '500.493.492',
    '371.370',
    '381.378',
    '1.177',
    '150.381.379',
    '1.339',
    '1.377',
    '148.365',
    '152.382',
    '59.62.378',
    '1.161.4.359.356',
    '1.161.4.359.380',
    '1.161.4.359.354',
    '1.474.472',
    '475.473',
    '487.469',
    '450.380',
    '353.382',
    '1.374',
    '1.370',
    '372.377',
    '466.380',
    '495.490',
    '500.493.492.491',
    '502.496.477.473',
    '385.384',
    '388.387',
    '392.391',
    '393.392.391',
    '393.392.394',
    '393.392.395',
    '393.392.396',
    '401.399',
    '406.405',
    '408.407',
    '416.413',
    '420.417',
    '423.421',
    '426.424',
    '430.428',
    '433.431',
    '150.381.380',
    '58.380',
    '1.161.381',
    '1.294.377',
    '1.161.4.374',
    '1.161.4.359.357',
    '480.472',
    '481.471',
    '486.465',
    '153.134.177',
    '459.380',
    '1.161.491',
    '492.491.471',
    '372.378',
    '498.497',
    '499.495.490',
    '500.493.492.491.471',
    '503.498',
    '1.355',
    '1.161.4.355',
    '1.161.4.377',
    '1.161.4.359.377',
    '482.469',
    '152.316.380',
    '463.383',
    '1.161.491.471',
    '492.491',
    '493.492.491.471',
    '496.477.473',
    '502.496',
    '437.435',
    '440.438',
    '446.443',
    '148.77.80',
    '448.447',
    '452.451',
    '455.454',
    '457.456',
    '462.460',
    '463.462.460',
    '463.462.461',
    '463.464',
    '466.470',
    '497.494',
    '498.497.494',
    '502.496.477',
    '503.498.497.494',
    '59.441',
  ],
  success: true,
};
export const mockUpdateRankEvent = {
  result: null,
  success: true,
};
