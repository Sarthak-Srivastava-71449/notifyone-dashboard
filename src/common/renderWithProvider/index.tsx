export { default as withProvider } from './renderWithProvider';
export { store, withLocationProvider } from './renderWithProvider';
export {
  mockEmailEventDetails,
  mockPushNotificationEventDetails,
  mockSmsEventDetails,
  mockWhatsappEventDetails,
} from './constants';
