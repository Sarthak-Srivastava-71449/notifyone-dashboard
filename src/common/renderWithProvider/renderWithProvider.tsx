import React from 'react';
import thunk from 'redux-thunk';
import {
  applyMiddleware,
  combineReducers,
  legacy_createStore as createStore,
} from 'redux';
import { composeWithDevTools } from '@redux-devtools/extension';
import { Provider } from 'react-redux';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import { ToastContainer } from 'react-toastify';

import messages from '../i18n/messages';
import googleReducer from 'src/platform/reducers/auth.reducer';

import emailEventsReducer from 'src/store/reducers/emailEvents.reducer';
import pushNotificationEventsReducer from 'src/store/reducers/pushNotificationEvents.reducer';
import smsEventsReducer from 'src/store/reducers/smsEvents.reducer';
import whatsAppEventsReducer from 'src/store/reducers/whatsappEvents.reducer';

import Dashboard from 'src/view/components/Dashboard/Dashboard';
import CommunicationList from 'src/view/components/CommunicationList/CommunicationList';
import Communication from 'src/view/components/Communication/Communication';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // Deprecated
    removeListener: jest.fn(), // Deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock('antd', () => {
  const antd = jest.requireActual('antd');

  return {
    ...antd,
  };
});

jest.mock('common/src/components/Spinner', () => ({
  Spinner: () => {
    return <></>;
  },
}));

let store: any;

const createTestStore = () => {
  const store = createStore(
    combineReducers({
      emailEvents: emailEventsReducer,
      pushNotificationEvents: pushNotificationEventsReducer,
      smsEvents: smsEventsReducer,
      whatsAppEvents: whatsAppEventsReducer,
      user: googleReducer,
    }),
    composeWithDevTools(applyMiddleware(thunk))
  );
  return store;
};

const withProvider = (component: any) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>{component}</BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

const withLocationProvider = (to: string, state: any = {}) => {
  store = createTestStore();
  store.dispatch = jest.fn();
  const locale = 'en-us';

  window.history.pushState(state, '', to);

  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={messages[locale]}>
        <BrowserRouter>
          <ToastContainer />
          <Routes>
            <Route path="communication" element={<Dashboard />}>
              <Route path="templates" element={<CommunicationList />} />
              <Route path="template" element={<Communication />} />
            </Route>
          </Routes>
        </BrowserRouter>
      </IntlProvider>
    </Provider>
  );
};

export { store, withLocationProvider };
export default withProvider;
