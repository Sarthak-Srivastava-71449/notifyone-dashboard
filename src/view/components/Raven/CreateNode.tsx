import { Input, Select } from 'antd';
import React, { Fragment, useEffect, useState } from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { Button } from '../Button';
import { RootState } from 'src/store';
import { fetchRavenNodeActions } from 'src/store/actions/ravenActionNodeEvents';
import { fetchRavenNodes } from 'src/store/actions/ravenNodeEvents';
import { createNodeEvent } from 'src/store/actions/ravenRootNodeEvents';
import {
  getCreateNodeSuccess,
  getNewNodeOptions,
  getNodeActions,
} from 'src/store/selectors/createNode';
import { getAccessToken } from 'src/store/selectors/accessToken';
import './Raven.css';

const CreateNode: React.FC = () => {
  const [nodeName, setNodeName] = useState('');
  const [selectedNodeType, setSelectedNodeType] = useState('Select Node Type');
  const [selectedNode, setSelectedNode] = useState('');

  /*
   Redux Selectors for fetching data
 */

  const accessToken = useSelector((state: RootState) => getAccessToken(state));
  const createNodeSuccess = useSelector((state: RootState) =>
    getCreateNodeSuccess(state)
  );
  const newNodeOptions = useSelector((state: RootState) =>
    getNewNodeOptions(state)
  );
  const nodeActions = useSelector((state: RootState) => getNodeActions(state));

  const navigate = useNavigate();

  useEffect(() => {
    fetchData();
    if (createNodeSuccess === true) {
      navigate('/communication/raven', { replace: true });
    }
  }, [createNodeSuccess]);

  const { Option } = Select;
  const dispatch = useDispatch();
  const intl = useIntl();

  // fetch all the updated data
  const fetchData = () => {
    dispatch(fetchRavenNodes(accessToken, 200, 0));
    dispatch(fetchRavenNodeActions(accessToken, 200, 0));
  };

  useEffect(() => {
    let string = '';
    if (selectedNodeType === 'Sub Question') {
      string = 'Select Node';
    } else if (selectedNodeType === 'Node Action') {
      string = 'Select Node Action';
    } else {
      string = '';
    }
    setSelectedNode(string);
  }, [selectedNodeType]);

  // handler for node selection dropdown
  const handleChangeNode = (selectedNodeIdValue: string) => {
    setSelectedNode(selectedNodeIdValue);
  };

  // handler to create node submit button
  const createNodeHandler = () => {
    const payload = {
      application: 'PHARMACY',
      nodeName: nodeName.trim(),
      selectedNode: '',
      selectedNodeAction: '',
      selectedNodeType: selectedNodeType,
    };
    if (!parseInt(selectedNode)) {
      // Check if any Node ID is selected or not
      return;
    }
    if (selectedNodeType === 'Sub Question') {
      payload.selectedNode = selectedNode;
    } else if (selectedNodeType === 'Node Action') {
      payload.selectedNodeAction = selectedNode;
    }
    dispatch(createNodeEvent(accessToken, payload));
    setNodeName('');
    setSelectedNodeType('Select Node Type');
    setSelectedNode('');
  };

  return (
    <Fragment>
      <div className="mx-auto m-6 raven-wrap wallet-wrapper shadow-md">
        <h2 className="text-lg raven-form-title font-semibold text-center py-1">
          <FormattedMessage id="create_node_form" />
        </h2>
        <div className="px-3 py-2 flex-col flex">
          <div className="flex mb-3  gap-4">
            <div className="input-label flex-1">
              <h3
                data-testid="node_details_data"
                className="text-lg rounded-md  bg-blue-400   text-white px-1 font-semibold py-1"
              >
                <FormattedMessage id="node_details" />
              </h3>
              <div className="py-3">
                <span className="text-sm px-1 font-semibold">
                  <FormattedMessage id="enter_node_name" />
                  <span className="px-1">:</span>
                </span>
                <div>
                  <Input
                    className={
                      'w-full text-sm leading-6  rounded-md py-1  shadow-sm'
                    }
                    placeholder={intl?.formatMessage({
                      id: 'enter_node_name',
                    })}
                    type="text"
                    data-testid="enter_node_name"
                    onChange={(e) => setNodeName(e.target.value)}
                  />
                </div>
              </div>
            </div>
          </div>
          <div className="flex mb-3 gap-4">
            <div className="input-label flex-1">
              <h3 className="text-lg rounded-md  bg-blue-400   text-white px-1 font-semibold py-1">
                <FormattedMessage id="node_type" />
              </h3>
              <div className="py-3">
                <span className="text-sm px-1  font-semibold">
                  <FormattedMessage id="select_node_type" />
                  <span className="px-1">:</span>
                </span>
                <div>
                  <Select
                    placeholder={intl?.formatMessage({
                      id: 'select_node_type',
                    })}
                    value={selectedNodeType}
                    className="w-full text-sm leading-6  rounded-md shadow-sm"
                    onChange={(nodeType: string) =>
                      setSelectedNodeType(nodeType)
                    }
                  >
                    <Option value="Select Node Type">
                      <FormattedMessage id="select_node_type" />
                    </Option>
                    <Option value="Sub Question">
                      <FormattedMessage id="sub_question" />
                    </Option>
                    <Option value="Node Action">
                      {' '}
                      <FormattedMessage id="node_action" />
                    </Option>
                  </Select>
                </div>
              </div>
            </div>
            {selectedNodeType !== 'Select Node Type' && (
              <div className="flex-1">
                <h3 className="text-lg rounded-md  bg-blue-400   text-white px-1 font-semibold py-1">
                  {selectedNodeType === 'Sub Question' && (
                    <FormattedMessage id="node_details" />
                  )}
                  {selectedNodeType === 'Node Action' && (
                    <FormattedMessage id="node_action_details" />
                  )}
                </h3>
                <div className="py-3">
                  <span className="text-sm px-1 font-semibold">
                    {selectedNodeType === 'Sub Question' && (
                      <FormattedMessage id="select_node" />
                    )}
                    {selectedNodeType === 'Node Action' && (
                      <FormattedMessage id="select_node_action" />
                    )}
                    <span className="px-1">:</span>
                  </span>

                  <div>
                    <Select
                      placeholder={
                        selectedNodeType === 'Sub Question'
                          ? intl.formatMessage({
                              id: 'select_node',
                            })
                          : selectedNodeType === 'Node Action'
                          ? intl?.formatMessage({
                              id: 'select_node_action',
                            })
                          : ''
                      }
                      value={selectedNode}
                      className="w-full text-sm leading-6  rounded-md shadow-sm"
                      onChange={handleChangeNode}
                    >
                      {selectedNodeType === 'Sub Question' &&
                        newNodeOptions?.length > 0 &&
                        newNodeOptions.map((item: any) => (
                          <Option
                            key={item.id}
                            value={item?.id}
                          >{`${item?.name}`}</Option>
                        ))}
                      {selectedNodeType === 'Node Action' &&
                        nodeActions?.length > 0 &&
                        nodeActions.map((item: any) => (
                          <Option
                            key={item.id}
                            value={item?.id}
                          >{`${item?.header}`}</Option>
                        ))}
                    </Select>
                  </div>
                </div>
              </div>
            )}
          </div>
          <div className="flex justify-center items-center">
            <Button
              data-testid="node_Submit_button"
              className=""
              onClick={createNodeHandler}
              disabled={!parseInt(selectedNode) || nodeName?.trim().length < 1}
            >
              <FormattedMessage id="submit" />
            </Button>
          </div>
        </div>
      </div>
    </Fragment>
  );
};
export default React?.memo(CreateNode);
