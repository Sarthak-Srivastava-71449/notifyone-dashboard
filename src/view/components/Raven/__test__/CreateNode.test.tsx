import React from 'react';
import { withProvider } from '../../../../common/renderWithProvider';
import '@testing-library/jest-dom/extend-expect';
import * as redux from 'react-redux';
import {
  render,
  cleanup,
  getByTestId,
  fireEvent,
  screen,
  waitFor,
} from '@testing-library/react';
import CreateNode from '../CreateNode';

import userEvent from '@testing-library/user-event';
import {
  accessToken,
  createNodeSuccess,
  newNodeOptions,
  nodeActions,
} from './mock';

jest.mock('src/store/actions/ravenActionNodeEvents');
jest.mock('src/store/actions/ravenNodeEvents');
jest.mock('src/store/actions/ravenRootNodeEvents');
// Raven Create Node  Test Cases

describe('Raven Create-Node-Test Suite', () => {
  jest.fn(() => Promise.resolve());
  beforeEach(() => {
    jest.clearAllMocks();
  });
  afterEach(cleanup);
  test('Match Create Node Snapshot', async () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => accessToken)
      .mockImplementationOnce(() => createNodeSuccess)
      .mockImplementationOnce(() => newNodeOptions)
      .mockImplementationOnce(() => nodeActions);
    const { asFragment } = render(withProvider(<CreateNode />));
    expect(asFragment()).toMatchSnapshot();
  });

  test('Showcase Mock Data', async () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => accessToken)
      .mockImplementationOnce(() => createNodeSuccess)
      .mockImplementationOnce(() => newNodeOptions)
      .mockImplementationOnce(() => nodeActions);

    render(withProvider(<CreateNode />));
  });

  test('React Internationalization - Should Work As Expected', () => {
    const component = render(withProvider(<CreateNode />));
    expect(component.container).toHaveTextContent('Create Node Form');
  });

  it('with only node name', async () => {
    const { getByTestId } = render(withProvider(<CreateNode />));
    fireEvent.change(getByTestId('enter_node_name'), {
      target: { value: 'enter_node_name' },
    });
    fireEvent.click(getByTestId('node_Submit_button'));
  });

  test('Select should change dropdown', async () => {
    const component = render(withProvider(<CreateNode />));
    const selectNodeDropdown = component.getAllByRole('combobox')[0];
    await userEvent.click(selectNodeDropdown);
    await waitFor(() => screen.getAllByText('Node Action')[0]);
    fireEvent.click(screen.getAllByText('Node Action')[0]);
    await waitFor(() =>
      expect(component.container).toHaveTextContent('Node Action')
    );
  });

  test('Testing Select DropDown Functionality', async () => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => accessToken)
      .mockImplementationOnce(() => createNodeSuccess)
      .mockImplementationOnce(() => newNodeOptions)
      .mockImplementationOnce(() => nodeActions);
    const component = render(withProvider(<CreateNode />));
    const selectDropDown = component.getAllByRole('combobox');
    await userEvent.click(selectDropDown[0]);
    await waitFor(() => screen.getAllByText('Sub Question')[0]);
    fireEvent.click(screen.getAllByText('Sub Question')[0]);
  });
});
