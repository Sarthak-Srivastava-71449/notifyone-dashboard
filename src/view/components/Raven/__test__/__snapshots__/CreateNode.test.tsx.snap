// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Raven Create-Node-Test Suite Match Create Node Snapshot 1`] = `
<DocumentFragment>
  <div
    class="mx-auto m-6 raven-wrap wallet-wrapper shadow-md"
  >
    <h2
      class="text-lg raven-form-title font-semibold text-center py-1"
    >
      Create Node Form
    </h2>
    <div
      class="px-3 py-2 flex-col flex"
    >
      <div
        class="flex mb-3  gap-4"
      >
        <div
          class="input-label flex-1"
        >
          <h3
            class="text-lg rounded-md  bg-blue-400   text-white px-1 font-semibold py-1"
            data-testid="node_details_data"
          >
            Node Details
          </h3>
          <div
            class="py-3"
          >
            <span
              class="text-sm px-1 font-semibold"
            >
              Enter Node Name
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <div>
              <input
                class="ant-input w-full text-sm leading-6  rounded-md py-1  shadow-sm"
                data-testid="enter_node_name"
                placeholder="Enter Node Name"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex mb-3 gap-4"
      >
        <div
          class="input-label flex-1"
        >
          <h3
            class="text-lg rounded-md  bg-blue-400   text-white px-1 font-semibold py-1"
          >
            Node Type
          </h3>
          <div
            class="py-3"
          >
            <span
              class="text-sm px-1  font-semibold"
            >
              Select Node Type
              <span
                class="px-1"
              >
                :
              </span>
            </span>
            <div>
              <div
                class="ant-select w-full text-sm leading-6  rounded-md shadow-sm ant-select-single ant-select-show-arrow"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-item"
                  >
                    Select Node Type
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="flex justify-center items-center"
      >
        <button
          class="button primary-button "
          data-testid="node_Submit_button"
          disabled=""
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
